src/libs/actions/Report.ts

```diff
@@ -3278,6 +3278,10 @@ function openReportFromDeepLink(url: string) {
             const connection = Onyx.connect({
                 key: ONYXKEYS.NVP_ONBOARDING,
                 callback: (val) => {
+                    if (!val) {
+                        return;
+                    }
+
```

Comment:
Anonymous users should be able to deep link without the need to wait for the onboarding value.

This change caused a regression when accessing public rooms as an anonymous user. The added delay made it that we wait for the `OpenApp` call which clears the optimistic public room report.

Coming from https://github.com/Expensify/App/issues/63675